packages/widgets/src/commandpalette.ts

```diff
@@ -973,6 +980,41 @@ export namespace CommandPalette {
       return kb ? CommandRegistry.formatKeystroke(kb.keys) : null;
     }
 
+    /**
+     * @returns The aria label content to add to the shortcut node.
+     */
+    formatItemAria(data: IItemRenderData): h.Child {
+      const keyToText: { [key: string]: string } = {
+        ']': 'Closing bracket',
+        '[': 'Opening bracket',
+        ',': 'Comma',
+        '.': 'Full stop',
+        "'": 'Single quote',
+        '-': 'Hyphen-minus'
+      };
```

Comment:
I think that we would want to make it easy to override in subclasses of the renderer so that they could reuse the logic below, but change the values on the right-hand side. In particular, JupyterLab will want to substituted the values with translated versions.

One solution would be making `keyToText` an option that can be passed to the renderer constructor, something like:

```ts
constructor(options: IRenderer.IOptions = {}) {
  this._keyToText = options.keyToText ?? DEFAULT_KEY_TO_TEXT;
}
```

But please read the next comment which tackles the deeper question on where the logic should be placed before implementing the suggestion above.

Reply:
@krassowski I had tried to use aria-keyshortcuts in place of aria-label however, my screen reader (Windows Narrator) still omits some punctuation. With aria-label it is read out as expected.

I have implemented your other suggestions which should now allow translations to be substituted in JupyterLab